utils/browser.py

```diff
@@ -229,7 +229,25 @@ def ensure_open(self, url_key=None):
             self.start(url_key=url_key)
         return self.browser
 
+    def add_cleanup(self, callback):
+        assert self.browser is not None
+        try:
+            cl = self.browser.__cleanup
+        except AttributeError:
+            cl = self.browser.__cleanup = []
```

Comment:
Why not just make the browser object have a __cleanup = None in the constructor and why use dunder?